scipy/stats/_continuous_distns.py

```diff
@@ -6294,6 +6294,62 @@ def _stats(self, a):
     def _entropy(self, a):
         return 1 - 1.0/a - np.log(a)
 
+    def fit(self, data, *args, **kwds):
+        data, fshape, floc, fscale = _check_fit_input_parameters(self, data,
+                                                                 args, kwds)
+        if floc is not None:
+            # `floc` is fixed, `fshape` and `fscale` may be deterermined
+            # through analytical equations
+            if fscale is None:
+                fscale = np.max(data) - floc
+            if fshape is None:
+                fshape = - len(data) / (np.sum(np.log((data - floc)/fscale)))
+            return fshape, floc, fscale
+
+        elif fscale is not None:
+            # `floc` can be determined by reording the equation for `fscale`
+            floc = np.max(data) - fscale
+            if fshape is None:
+                fshape = - len(data) / (np.sum(np.log((data - floc)/fscale)))
+            return fshape, floc, fscale
+
+        elif fshape is not None:
+            # only fshape is fixed, optimize over `floc` using dependent
+            # `fscale` equation and compare resulting calculation of `fshape`
+            # to actual `fshape`
+            def func(floc):
+                fscale = np.max(data) - floc
+                # subtract actual `fshape` to calculated `fshape`
+                return (- len(data) / (np.sum(np.log((data - floc) /
```

Comment:
this condition is not correct when the shape is fixed

Reply:
@mdhaber

Fixing this to use the correct equation appears to fix the failure I was getting when only `fshape` was fixed.